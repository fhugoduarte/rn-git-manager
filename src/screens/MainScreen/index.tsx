import React, { useEffect, useState, Fragment } from 'react';
import { Animated } from 'react-native';
import MCIcon from 'react-native-vector-icons/MaterialCommunityIcons';
import FWIcon from 'react-native-vector-icons/FontAwesome';

import ModalLogin from '../../components/UI/ModalLogin';
import themes from '../../utils/themes';
import {
  Container,
  Title,
  TitleContainer,
  BannerContainer,
  ButtonsContainer,
  ButtonsContainerTitle,
  SignBitBucketButton,
  SignGitButton,
  SkipButton,
  SignButtonTitle,
} from './styles';

enum gitType {
  bitBucket = 'bitbucket',
  gitHub = 'github',
}

interface State {
  open: boolean;
  git?: gitType;
}

const mainScreen: React.FC<{}> = () => {
  let animatedValue = new Animated.Value(0);
  const [modalLoginState, changeModalLoginState] = useState<State>({
    open: false,
    git: gitType.gitHub,
  });

  const openLoginModal = (git: gitType): void => {
    changeModalLoginState({ open: true, git });
  };

  const closeLoginModal = (): void => {
    changeModalLoginState({ ...modalLoginState, open: false });
  };

  useEffect((): void => {
    Animated.timing(animatedValue, {
      toValue: 100,
      duration: 1000,
      useNativeDriver: true,
    }).start();
  }, []);

  return (
    <Fragment>
      <ModalLogin
        visible={modalLoginState.open}
        git={modalLoginState.git}
        onClose={closeLoginModal}
      />
      <Container
        style={{
          opacity: animatedValue.interpolate({
            inputRange: [0, 100],
            outputRange: [0, 1],
          }),
        }}
      >
        <BannerContainer
          style={{
            transform: [
              {
                translateY: animatedValue.interpolate({
                  inputRange: [0, 100],
                  outputRange: [-100, 0],
                  extrapolate: 'clamp',
                }),
              },
            ],
          }}
        >
          <MCIcon name="git" size={100} color={themes.gitColor} />
          <TitleContainer>
            <Title>Git</Title>
            <Title>Manager</Title>
          </TitleContainer>
        </BannerContainer>
        <ButtonsContainer
          style={{
            transform: [
              {
                translateY: animatedValue.interpolate({
                  inputRange: [0, 100],
                  outputRange: [100, 0],
                  extrapolate: 'clamp',
                }),
              },
            ],
          }}
        >
          <ButtonsContainerTitle>Entrar com:</ButtonsContainerTitle>
          <SignGitButton onPress={(): void => openLoginModal(gitType.gitHub)}>
            <FWIcon name="github" size={40} color="#FFF" />
            <SignButtonTitle>GitHub</SignButtonTitle>
          </SignGitButton>
          <SignBitBucketButton
            onPress={(): void => openLoginModal(gitType.bitBucket)}
          >
            <FWIcon name="bitbucket" size={40} color="#FFF" />
            <SignButtonTitle>BitBucket</SignButtonTitle>
          </SignBitBucketButton>
        </ButtonsContainer>
        <SkipButton>
          <SignButtonTitle>Continuar sem Logar</SignButtonTitle>
        </SkipButton>
      </Container>
    </Fragment>
  );
};

export default mainScreen;
