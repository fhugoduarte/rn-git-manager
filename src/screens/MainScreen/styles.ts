import styled from 'styled-components/native';

import themes from '../../utils/themes';

const Container = styled.View`
  flex: 1;
  background-color: ${themes.lightGray};
  align-items: center;
  justify-content: center;
`;

const BannerContainer = styled.View`
  flex: 0.2;
  flex-direction: row;
  justify-content: center;
  align-items: center;
`;

const GitLogo = styled.Image`
  width: 50px;
  height: 50px;
`;

const TitleContainer = styled.View`
  padding: 0px 10px;
`;

const Title = styled.Text`
  font-weight: bold;
  font-size: 35px;
`;

const ButtonsContainer = styled.View`
  padding: 30px 20px;
  flex: 0.6;
`;

const ButtonsContainerTitle = styled.Text`
  font-weight: bold;
  font-size: 30;
  align-self: center;
  padding-bottom: 30px;
`;

const SignButton = styled.TouchableOpacity`
  border-radius: 4px;
  margin-top: 20px;
  padding: 10px 40px;
  align-items: center;
  flex-direction: row;
`;

const SignGitButton = styled(SignButton)`
  background-color: ${themes.gitHub};
`;

const SignBitBucketButton = styled(SignButton)`
  background-color: ${themes.bitBucket};
`;

const SignButtonImage = styled.Image`
  width: 40px;
  height: 40px;
`;

const SignButtonTitle = styled.Text`
  color: #fff;
  font-size: 25px;
  font-weight: 400;
  padding-left: 15px;
`;

export {
  Container,
  TitleContainer,
  Title,
  BannerContainer,
  GitLogo,
  ButtonsContainer,
  ButtonsContainerTitle,
  SignGitButton,
  SignBitBucketButton,
  SignButtonTitle,
  SignButtonImage,
};
