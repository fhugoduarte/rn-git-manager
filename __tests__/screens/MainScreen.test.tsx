import React from 'react';
import renderer from 'react-test-renderer';

import MainScreen from '../../src/screens/MainScreen';

jest.mock('NativeAnimatedHelper');
jest.useFakeTimers();

describe('Render MainScreen', (): void => {
  it('should renders correctly', (): void => {
    const tree = renderer.create(<MainScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
