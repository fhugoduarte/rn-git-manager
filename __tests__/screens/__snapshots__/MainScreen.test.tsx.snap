// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render MainScreen should renders correctly 1`] = `
Array [
  <View
    style={
      Object {
        "alignItems": "center",
        "height": 1334,
        "justifyContent": "center",
        "opacity": 0,
        "position": "absolute",
        "width": 750,
      }
    }
  >
    <View
      accessible={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#00000040",
          "height": 1334,
          "position": "absolute",
          "width": 750,
        }
      }
    />
    <View
      onLayout={[Function]}
      style={
        Array [
          Array [
            Object {
              "alignItems": "center",
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "justifyContent": "center",
              "position": "absolute",
            },
          ],
          Object {
            "paddingBottom": 0,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "white",
            "borderColor": "gray",
            "borderRadius": 6,
            "borderWidth": 2,
            "justifyContent": "center",
            "paddingBottom": 20,
            "paddingLeft": 20,
            "paddingRight": 20,
            "paddingTop": 20,
            "transform": Array [
              Object {
                "translateY": 100,
              },
            ],
            "width": 600,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "paddingBottom": 10,
                "paddingLeft": 10,
                "paddingRight": 10,
                "paddingTop": 10,
              },
            ]
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "#333",
                  "fontSize": 50,
                },
                undefined,
                Object {
                  "fontFamily": "FontAwesome",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
        <Text
          style={
            Array [
              Object {
                "fontSize": 18,
                "fontWeight": "bold",
                "paddingBottom": 15,
                "paddingLeft": 5,
                "paddingRight": 5,
                "paddingTop": 15,
              },
            ]
          }
        >
          Usuário ou Email
        </Text>
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          rejectResponderTermination={true}
          style={
            Array [
              Object {
                "borderColor": "gray",
                "borderRadius": 3,
                "borderWidth": 1,
                "fontSize": 18,
                "paddingBottom": 10,
                "paddingLeft": 10,
                "paddingRight": 10,
                "paddingTop": 10,
              },
            ]
          }
          underlineColorAndroid="transparent"
          value=""
        />
        <Text
          style={
            Array [
              Object {
                "fontSize": 18,
                "fontWeight": "bold",
                "paddingBottom": 15,
                "paddingLeft": 5,
                "paddingRight": 5,
                "paddingTop": 15,
              },
            ]
          }
        >
          Senha
        </Text>
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          rejectResponderTermination={true}
          style={
            Array [
              Object {
                "borderColor": "gray",
                "borderRadius": 3,
                "borderWidth": 1,
                "fontSize": 18,
                "paddingBottom": 10,
                "paddingLeft": 10,
                "paddingRight": 10,
                "paddingTop": 10,
              },
            ]
          }
          underlineColorAndroid="transparent"
          value=""
        />
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "justifyContent": "center",
              },
            ]
          }
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#2ebc4f",
                "borderRadius": 4,
                "flexDirection": "row",
                "justifyContent": "center",
                "marginTop": 20,
                "opacity": 1,
                "paddingBottom": 10,
                "paddingLeft": 40,
                "paddingRight": 40,
                "paddingTop": 10,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#fff",
                    "fontSize": 18,
                    "fontWeight": "400",
                    "paddingLeft": 15,
                  },
                ]
              }
            >
              Entrar
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>,
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#f5f5f5",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "space-around",
        "opacity": 0,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "center",
          "transform": Array [
            Object {
              "translateY": -100,
            },
          ],
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#f1502f",
              "fontSize": 60,
            },
            undefined,
            Object {
              "fontFamily": "Material Design Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
      <View
        style={
          Array [
            Object {
              "paddingBottom": 0,
              "paddingLeft": 10,
              "paddingRight": 10,
              "paddingTop": 0,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontSize": 23,
                "fontWeight": "bold",
              },
            ]
          }
        >
          Git
        </Text>
        <Text
          style={
            Array [
              Object {
                "fontSize": 23,
                "fontWeight": "bold",
              },
            ]
          }
        >
          Manager
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "paddingBottom": 10,
          "paddingLeft": 20,
          "paddingRight": 20,
          "paddingTop": 10,
          "transform": Array [
            Object {
              "translateY": 100,
            },
          ],
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "alignSelf": "center",
              "fontSize": 18,
              "fontWeight": "bold",
              "paddingBottom": 10,
            },
          ]
        }
      >
        Entrar com:
      </Text>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#333",
            "borderRadius": 4,
            "flexDirection": "row",
            "marginTop": 20,
            "opacity": 1,
            "paddingBottom": 10,
            "paddingLeft": 20,
            "paddingRight": 20,
            "paddingTop": 10,
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#FFF",
                "fontSize": 30,
              },
              undefined,
              Object {
                "fontFamily": "FontAwesome",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#fff",
                "fontSize": 18,
                "fontWeight": "400",
                "paddingLeft": 15,
              },
            ]
          }
        >
          GitHub
        </Text>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#205081",
            "borderRadius": 4,
            "flexDirection": "row",
            "marginTop": 20,
            "opacity": 1,
            "paddingBottom": 10,
            "paddingLeft": 20,
            "paddingRight": 20,
            "paddingTop": 10,
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#FFF",
                "fontSize": 30,
              },
              undefined,
              Object {
                "fontFamily": "FontAwesome",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#fff",
                "fontSize": 18,
                "fontWeight": "400",
                "paddingLeft": 15,
              },
            ]
          }
        >
          BitBucket
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#6cc644",
          "borderRadius": 4,
          "flexDirection": "row",
          "marginTop": 20,
          "opacity": 1,
          "paddingBottom": 10,
          "paddingLeft": 20,
          "paddingRight": 20,
          "paddingTop": 10,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#fff",
              "fontSize": 18,
              "fontWeight": "400",
              "paddingLeft": 15,
            },
          ]
        }
      >
        Continuar sem Logar
      </Text>
    </View>
  </View>,
]
`;
